packages/sources/coinmarketcap/src/transport/crypto.ts

```diff
@@ -79,14 +82,14 @@ export const httpTransport = new HttpTransport<HttpTransportTypes>({
       }
     }
 
-    for (const [idType, fullList] of Object.entries(groupedParams)) {
-      if (fullList && fullList.length > 0) {
-        // CMC does not support more than 120 unique quotes
-        const chunkedList = splitArrayIntoChunks(fullList, 120)
-
-        // This could be further optimized in cases with more than 120 entries, to make sure that
-        // chunkes are grouped optimally to avoid sending unnecessary converts
-        for (const list of chunkedList) {
+    for (const [idType, list] of Object.entries(groupedParams)) {
+      if (list && list.length > 0) {
+        const batchedBaseCurrencies = [...new Set(list.map((p) => p.cid || p.slug || p.base))].join(
+          ',',
+        )
+        // We don't want to batch quotes but only base currencies, so for each unique quote in each list we create a request
+        // This could be further optimized to make sure that lists are grouped optimally to avoid sending unnecessary bases
+        ;[...new Set(list.map((p) => p.quote.toUpperCase()))].forEach((uniqueQuote) => {
```

Comment:
Sorry should have mentioned this earlier but for readability maybe we should separate this in to 2 lines? like
```suggestion
        const uniqueQuotes = [...new Set(list.map((p) => p.quote.toUpperCase()))]
        uniqueQuotes.forEach((uniqueQuote) => {
```